Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java

```diff
@@ -472,8 +476,12 @@ void switchStructure(Object treeNodeData) throws InvalidMetadataValueException,
         }
     }
 
-    void switchMediaUnit() throws InvalidMetadataValueException, NoSuchMetadataFieldException {
-        metadataPanel.preservePhysical();
+    void switchMediaUnit() throws NoSuchMetadataFieldException {
+        try {
+            metadataPanel.preservePhysical();
+        } catch (InvalidMetadataValueException e) {
+            logger.info(e.getLocalizedMessage());
+        }
         metadataPanel.showPhysical(structurePanel.getSelectedMediaUnit());
```

Comment:
I know this part of code is not changed but in case you would have time to change you could assign _structurePanel.getSelectedStructure()_ to variable. It would save us false positive messages that _get()_ is used without _isPresent()_ call.

Reply:
The `isPresent` check is done inside the `showPhysical` method. Are you suggesting to change that method so it expects parameters of type `MediaUnit` instead of `Optional<MediaUnit>`?

Reply:
Next check is done in this method:
```
        if (structurePanel.getSelectedMediaUnit().isPresent()) {
            // update gallery
            galleryPanel.updateSelection(structurePanel.getSelectedMediaUnit().get());
```